# 02. 자료형


## 02-1 숫자형


### 숫자형은 어떻게 만들고 사용할까?


#### 정수형


정수형(integer)은 말 그대로 정수를 뜻하는 자료형이다. 다음은 양의 정수, 음의 정수, 숫자 0을 변수 a에 대입하는 예이다.


In [1]:
a = 123
print(a)
a = -178
print(a)
a = 0
print(a)

123
-178
0


#### 실수형


실수형(floating-point)은 소수점이 포함된 숫자를 말한다. 다음은 실수를 변수 a에 대입하는 예이다.


In [2]:
a = 1.2
print(a)
a = -3.45
print(a)

1.2
-3.45


정수 1과 실수 1.0은 값은 같지만 자료형이 다르다.


아주 크거나 아주 작은 숫자를 표현할 때는 지수 표현 방식을 사용하면 편리하다. 파이썬에서는 4.24e10 또는 4.24E10처럼 표현할 수 있다(e와 E 둘 중 어느 것을 사용해도 된다).


In [4]:
a = 4.24e10
print(a)
a = 4.24e-10
print(a)

42400000000.0
4.24e-10


#### 8진수와 16진수


8진수(octal)는 0o 또는 0O(숫자 0 + 알파벳 소문자 o 또는 대문자 O)로 시작한다.


In [6]:
a = 0o177
print(a)

127


$ 0o177 = 1\times 8^2 + 7\times 8^1 + 7 = 127 $


16진수(hexadecimal)는 0x로 시작한다.


In [8]:
a = 0x8FF
print(a)
b = 0xABC
print(b)

2303
2748


$ 0x8FF = 8\times 16^2 + 15\times 16^1 + 15 = 2303 \\
0xABC = 10\times 16^2 + 11\times 16^1 + 12 = 2748
$

A, B, C, D, E, F는 10, 11, 12, 13, 14, 15를 의미한다.


8진수나 16진수는 파이썬에서 자주 사용하지 않으므로 간단히 눈으로만 익히고 넘어가자. 다만 파일 권한이나 색상 코드 등에서 가끔 볼 수 있으므로 이런 표현이 있다는 정도만 기억해 두자.


### 숫자형을 활용하기 위한 연산자


#### 사칙 연산


In [12]:
a, b = 3, 4
print(a + b)
print(a - b)
print(a * b)
print(a / b)

7
-1
12
0.75


#### x의 y 제곱을 나타내는 \*\* 연산자


In [13]:
a, b = 3, 4
print(a**b)

81


#### 나눗셈 후 나머지를 반환하는 % 연산자


In [14]:
print(7 % 3)  # 1
print(3 % 7)  # 3

1
3


#### 나눗셈 후 몫을 반환하는 // 연산자


In [15]:
print(7 / 4)
print(7 // 4)

1.75
1


#### 복합 연산자


In [16]:
a = 1
print(a)
a = a + 1
print(a)

1
2


In [17]:
a = 1
print(a)
a -= 1
print(a)

1
0


In [18]:
a = 2
print(a)
a *= 3
print(a)

2
6
